compiler/rustc_serialize/src/serialize.rs

```diff
@@ -229,9 +229,9 @@ impl<D: Decoder, T> Decodable<D> for PhantomData<T> {
     }
 }
 
-impl<D: Decoder, T: Decodable<D>> Decodable<D> for Box<[T]> {
-    fn decode(d: &mut D) -> Box<[T]> {
-        let v: Vec<T> = Decodable::decode(d);
+impl<D: Decoder, A: std::alloc::Allocator + Default, T: Decodable<D>> Decodable<D> for Box<[T], A> {
```

Comment:
`use std::alloc::Allocator` earlier, then just do `Allocator` here since you mention it several times.